genome/gene_statistic.py

```
@@ -0,0 +1,454 @@
+                        _table2[c] = a, len(cs2), (gc_rank[cgc[c]] / len(cgc))
+                    sf_aa_codon.setdefault(len(cs2), {}).setdefault(a, []).append(cs2)
+        self.codon_table: Final = {
+            k: self.CodonFeat(*v, k[:2]) for k, v in _table2.items()
+        }
+        self.aa2codon: Final = aa2codon
+        self.stop_codons: Final = frozenset(stop_codons)
+        # pseudo codon family
+        self.pcf: Final = sf_aa_codon
+
+    @classmethod
+    def get(cls, table: int | str = 11, _table_cache={}) -> "CodonTable":
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)